src/plugins/intel_cpu/src/gemm/ov_cpu_gemm.h

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <cstddef>
+#include <cstdint>
+
+/**
+ * @brief  Computes the length in bytes for the packed matrix B buffer(SGEMM).
+ *
+ * @param N       Supplies the number of columns of matrix B.
+ * @param K       Supplies the number of rows of matrix B.
+ * @return        bytes of the packing buffer
+ */
+size_t ov_sgemm_pack_get_size(const int64_t N, const int64_t K);
+
+/**
+ * @brief  Packs the contents of matrix B
+ *
+ * @param transb  T for transpose B, N for none-tranpose B
+ * @param N       Supplies the number of columns of matrix B and matrix C.
+ * @param K       Supplies the number of columns of matrix A and the number
+                  of rows of matrix B.
+ * @param ldb     Supplies the first dimension of matrix B.
+ * @param src     Supplies the address of matrix B
+ * @param dst     Supplies pointer to prePacked B buffer
+ */
+void ov_sgemm_pack(const char* transb,
+                   const int64_t N,
+                   const int64_t K,
+                   const int64_t ldb,
+                   const float* src,
+                   float* dst);
+
+/**
+ * @brief  Signle threaded SGEMM
```

Comment:
it's used for mha custom node to calculate the gemm

Reply:
Got it, but what if we want to introduce the most unformal call for sgemm, it will be called `ov_sgemm`?
Why `compute` postfix is used? This does make sense in the case of prepacked matrix, since there are two stages: prepack and compute. But this call signature looks more like a typical BLAS `sgemm`. I'm not against keeping this `compute` postfix, I simply want to understand the motivation behind this naming convention.

Reply:
I just want to align the postfix the name between FC and Matmul. You are right, it's better to simplify both to `ov_sgemm` and `ov_packed_sgemm`

Reply:
I think we can use the same name convention for BLAS extensions as in MKL https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-fortran/2023-0/gemm-compute-002.html
That is `ov_sgemm_pack` `ov_sgemm_compute` and `ov_sgemm` when there is no prepacking. However, we probably also need `ov_sgemm` which utilize all available threads, and, though we do not need it right now, but we need to understand how we should distinguish single threaded `ov_sgemm` and multithreaded `ov_sgemm`.

Reply:
Maybe we could introduce an extra API like `set_sgemm_threads_num()` to control the threads before `ov_sgemm`

Reply:
Can we simply pass the threads number as a call parameter? With default value 0, which means all the available threads. And the MHA implementation will simply call it with parameter 1. What do you think?

Reply:
That's a good idea, the MLAS internally uses the pointer of ThreadPool as an indicator whether to use single-thread or the threadpool, I think we could use a `bool` to indicate single thread/multi-thread